Fixed server-client relationship

ttlaz123 opened the pull request:
Now, server correctly receives commands, and client correctly sends commands. Added a bunch of error handling and socket timeouts to prevent program from crashing under most commands. More testing is required to detect other areas where error handling is needed.

Fixed the do_wait() function so that pressing q will stop waiting. 

Potential improvements: allow more than one computer to connect and send commands, connect without having to type connect on the server.

ttlaz123 committed:
minor fixes

17lauraf committed:
fixed loop exiting

17lauraf committed:
added some error handling

17lauraf committed:
Update mycmd_1.py

17lauraf committed:
Update mycmd_2.py

17lauraf committed:
made sure the two files are the same

why are there even two different files in the first place?

ttlaz123 committed:
an attempt at fixing the while loop

will probably revert back to old way

ttlaz123 committed:
Update ntpTest.py

ttlaz123 committed:
swapped send and receive

ttlaz123 committed:
Delete mycmd_2.py

ttlaz123 committed:
cleanup

ttlaz123 committed:
fixed the waiting command

ttlaz123 committed:
Update mycmd_1.py

ttlaz123 committed:
Update mycmd_1.py

ttlaz123 committed:
Update mycmd_1.py

ttlaz123 committed:
Update mycmd_1.py

ttlaz123 committed:
Update mycmd_1.py

ttlaz123 committed:
Update mycmd_1.py

ttlaz123 committed:
Update mycmd_1.py

ttlaz123 committed:
fixed gather time

ttlaz123 committed:
fixed typo

ttlaz123 committed:
fixed bugs

shu-xiao commented on HLshell/mycmd_1.py:
```diff
@@ -71,13 +78,14 @@ def _checkInit(func):
+            except Exception as e:
                 ''' len(params)==0 '''
-                raise Exception('Syntax error: cannot check Initialization')
+                print(e)
```
should use print_error

ttlaz123 replied:
fixed

shu-xiao commented on HLshell/mycmd_1.py:
```diff
@@ -100,13 +113,13 @@ def do_exit(self, par):
     def default(self, inp):
-        if inp == 'q' or bool(match(inp,'qqq+')):
-            return self.do_exit()
+        if inp == 'exit' or inp == 'qqq':
```
I prefer typing 'q' or many q to leave. If program crashes, it may receive many q.
The final decision depends on you, choose one and delete the other.

ttlaz123 replied:
put the bool match back in

shu-xiao commented on HLshell/mycmd_1.py:
```diff
@@ -378,33 +403,49 @@ def __init__(self):
         pass
         #print('Open Server Shell')
 
     def do_connect(self,par):
```
If function requires more than 1 arg, add my syntax candy.
@parser()
def do_connect(self,*par):

ttlaz123 replied:
added

shu-xiao commented on alicptfts/alicptfts.py:
```diff
@@ -231,6 +232,45 @@ def generate_headers(self, scan_params, time_start, time_end):
+        calculates the number of seconds it would take to move from 
+        one end of range to the other end of range
+        '''
+        vf = np.sqrt(accel*max_range)
```
you can use math.sqrt. I avoid to use numpy since some old machine does not install it.
and also I suggest that adding type check
`vf = float(np.sqrt(accel*max_range))`

ttlaz123 replied:
fixed

shu-xiao commented on alicptfts/alicptfts.py:
```diff
@@ -231,6 +232,45 @@ def generate_headers(self, scan_params, time_start, time_end):
+        '''
+        vf = np.sqrt(accel*max_range)
+        if(vf < max_velocity):
+            max_time_one_pass = np.sqrt(max_range * 4 / accel)
```
like above. these parameters may all be int
`max_time_one_pass = np.sqrt(max_range * 4. / accel)`

ttlaz123 replied:
fixed, although I believe python automatically converts ints to floats when necessary

shu-xiao replied:
You are right. Python3 does it but Python2 doesn't.

shu-xiao commented on mynewportxps/newportxps/newportxps.py:
```diff
@@ -539,6 +539,8 @@ def set_velocity(self, stage, velo, accl=None,
             return
         ret, v_cur, a_cur, jt0_cur, jt1_cur = \
              self._xps.PositionerSGammaParametersGet(self._sid, stage)
+        if velo is None:
```
try not to change newportxps

ttlaz123 replied:
is this a hard rule? The only way around this would be to rewrite the entire function inside alicptfts, which I don't think is very useful. Is anyone else using the newportxps.py code? If not, I don't think it hurts to make a small change that also overall improves newportxps

shu-xiao reviewed (changes requested):
Some minor bugs need to be fixed

ttlaz123 committed:
fixed recommendations

ttlaz123 committed:
Merge remote-tracking branch 'upstream/main' into main

ttlaz123 committed:
made everything automatic

now the client continuously tries to connect to server

still have issues with receiving stderr

ttlaz123 committed:
Start function is functional

start function loops the client until we enter debug mode
